src/joypad/mod.rs

```diff
@@ -57,11 +69,35 @@ impl Joypad {
             action_select: false,
             direction_select: false,
             buttons: [false; 8],
+            int_req: false,
         }
     }
 
     pub fn update_button(&mut self, button: Button, state: bool) {
         let idx = button as usize;
+        let cur_state = self.buttons[idx];
+
+        // TODO: emulate switch bounce?
```

Comment:
I don't think we should try to emulated switch bounce. I think sdl+keyboard/controller driver will debounce inputs. Putting that back in would be difficult and unnecessary.

Reply:
Yes they debounce, but some games expect bouncing on there.  Not sure if any rely one it.  Testing will tell.